### PR TITLE
Created an icon which opens a new tab for SandDance 

### PR DESCRIPTION
**Pull request by RebeccaRunxinWang**

### CODE REVIEW

File: extensions/admin-tool-ext-win/package.json [hunk 1]
```diff
@@ -75,5 +75,10 @@
 	},
 	"devDependencies": {
 		"vscode": "1.0.1"
+	},
+	"__metadata": {
+		"id": "41",
+		"publisherDisplayName": "Microsoft",
```
Comment (Charles-Gagnon) [inline extensions/admin-tool-ext-win/package.json]: Undo the changes to this file

File: extensions/cms/package.json [hunk 2]
```diff
@@ -624,5 +624,10 @@
 		"should": "^13.2.1",
 		"vscode": "^1.1.26",
 		"typemoq": "^2.1.0"
+	},
+	"__metadata": {
+		"id": "40",
```
Comment (Charles-Gagnon) [inline extensions/cms/package.json]: undo the changes to this file

File: src/sql/workbench/parts/query/browser/actions.ts [hunk 3]
```diff
@@ -210,3 +210,19 @@ export class ChartDataAction extends Action {
 		return Promise.resolve(true);
 	}
 }
+
+export class SandDanceDataAction extends Action {
+	public static ID = 'grid.sandDance';
+	public static LABEL = localize('sandDance', 'SandDance');
+	public static ICON = 'viewSandDance';
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/actions.ts]: Should this really be localized? Normally product names aren't

File: src/sql/workbench/parts/query/browser/queryResultsView.ts [hunk 4]
```diff
@@ -169,6 +227,8 @@ export class QueryResultsView extends Disposable {
 		this.resultsTab = this._register(new ResultsTab(instantiationService));
 		this.messagesTab = this._register(new MessagesTab(instantiationService));
 		this.chartTab = this._register(new ChartTab(instantiationService));
+		//test
+		this.sandDanceTab = this._register(new SandDanceTab(instantiationService));
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryResultsView.ts]: Remove

File: src/sql/workbench/parts/query/browser/queryResultsView.ts [hunk 5]
```diff
@@ -204,9 +264,11 @@ export class QueryResultsView extends Disposable {
 		this.resultsTab.queryRunner = runner;
 		this.messagesTab.queryRunner = runner;
 		this.chartTab.queryRunner = runner;
+		this.sandDanceTab.queryRunner = runner; //test
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryResultsView.ts]: >//test [](start = 42, length = 6)

Remove comment

File: src/sql/workbench/parts/query/browser/queryResultsView.ts [hunk 6]
```diff
@@ -282,6 +350,8 @@ export class QueryResultsView extends Disposable {
 		this.topOperationsTab.view.state = this.input.state.topOperationsState;
 		this.chartTab.view.state = this.input.state.chartState;
 
+		[this.resultsTab, this.messagesTab, this.qpTab, this.topOperationsTab, this.chartTab, this.sandDanceTab].forEach(t => t.clear());
+
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryResultsView.ts]: Remove and update the one above the state assignments

File: src/sql/workbench/parts/query/browser/queryResultsView.ts [hunk 7]
```diff
@@ -326,12 +397,27 @@ export class QueryResultsView extends Disposable {
 		this.chartTab.chart(dataId);
 	}
 
+	public sandDanceData(dataId: { resultId: number, batchId: number }): void {
+		this.input.state.visibleTabs.add(this.sandDanceTab.identifier);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryResultsView.ts]: Would prefer that this and the other similar methods are named something like showSandDanceTab or showDataAsSandDance. It's more descriptive of what the function is actually doing.

File: src/sql/workbench/parts/query/browser/queryResultsView.ts [hunk 8]
```diff
@@ -103,6 +103,40 @@ class ResultsView extends Disposable implements IPanelView {
...
+	private container = document.createElement('div');
+
+	constructor(private instantiationService: IInstantiationService) {
+		super();
+	}
+
+	render(container: HTMLElement): void {
+		container.appendChild(this.container);
+		//this.container.innerText += "Hello World!";
+	}
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryResultsView.ts]: remove

File: src/sql/workbench/parts/query/browser/queryResultsView.ts [hunk 9]
```diff
@@ -147,12 +181,36 @@ class MessagesTab implements IPanelTab {
 	}
 }
 
+class SandDanceTab implements IPanelTab {
+	public readonly title = nls.localize('sandDanceTabTitle', 'SandDance');
+	public readonly identifier = 'sandDanceTab';
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryResultsView.ts]: same - does this need to be localized?

File: src/sql/workbench/parts/query/browser/queryResultsView.ts [hunk 10]
```diff
@@ -103,6 +103,40 @@ class ResultsView extends Disposable implements IPanelView {
 	}
 }
 
+class SandDanceView extends Disposable implements IPanelView {
+	private container = document.createElement('div');
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryResultsView.ts]: So right now this is just adding the button but it doesn't appear to actually display anything. What's the plan for implementing that behavior? You probably don't want to actually add the action until the tab will do something either

File: src/sql/base/browser/ui/table/media/slickColorTheme.css [hunk 11]
```diff
@@ -122,6 +122,11 @@
 	background-image: url("viewChart.svg");
 }
 
+.vs .icon.viewSandDance {
```
Comment (kburtram) [inline src/sql/base/browser/ui/table/media/slickColorTheme.css]: shouldn't this be from an extension?  Perhaps I missed what the planned feature is here.  Can we do a design review of what is happening here with the Query Editor committers?

**Comment by kburtram** [comment]
@RebeccaRunxinWang I'm removing the "Changes Requested" since @kevcunnane pointed out this PR is not against master.  If you're planning to merge to master at some point we'll need to do a design review.  My understanding was we're building an extension, so the changes to core should be generic extensibility points.  

It's fine to iterate as desired in feature branches.  But I'd recommend resolving core design concepts early to avoid have a lot of changes required in order to merge into master.
